geoh5py/objects/object_base.py

```diff
@@ -47,202 +48,60 @@ def __init__(self, object_type: ObjectType, **kwargs):
-            prop_group = [
-                pg for pg in self.property_groups if pg.name == prop_group.name
-            ][0]
+        date = datetime.now().strftime("%Y-%m-%dT%H:%M:%S")
+        if author is None:
+            author = ",".join(self.workspace.contributors)
+
+        comment_dict = {"Author": author, "Date": date, "Text": comment}
+
+        if self.comments is None:
+            self.add_data(
+                {
+                    "UserComments": {
+                        "values": [comment_dict],
+                        "association": "OBJECT",
+                        "entity_type": {"primitive_type": "TEXT"},
+                    }
+                }
+            )
         else:
-            self.property_groups = [prop_group]
-
-        return prop_group
-
-    def get_property_group(
-        self, group_id: Union[str, uuid.UUID]
-    ) -> Optional[PropertyGroup]:
-        """
-        Retrieve a property_group from one of its identifier, either by name or uuid
-
-        :param group_id: PropertyGroup identifier, either name or uuid
-
-        :return: PropertyGroup with the given name
-        """
-        if isinstance(group_id, uuid.UUID):
-            groups_list = [pg for pg in self.property_groups if pg.uid == group_id]
-
-        else:  # Extract all groups uuid with matching group_id
-            groups_list = [pg for pg in self.property_groups if pg.name == group_id]
-
-        try:
-            prop_group = groups_list[0]
-        except IndexError:
-            print(f"No property_group {group_id} found.")
-            return None
-
-        return prop_group
+            self.comments.values = self.comments.values + [comment_dict]
 
     def add_data(
         self, data: dict, property_group: str = None
     ) -> Union[Data, List[Data]]:
         """
-        Create data with association from dictionary of data objects name and values.
-        The provided values can either be a dictionary of kwargs accepted by the target
-        Data object class or an array of values. If not provided as argument, a data
-        association type is assigned based on the length of the given values.
+        Create :obj:`~geoh5py.data.data.Data` from dictionary of name and arguments.
+        The provided arguments can be any property of the target Data class.
+
+        :param data: Dictionary of data to be added to the object, e.g.
 
-        :param data: Dictionary of data to be added to the object,
-            e.g. data = {"data_name1": {'values', values1, 'association': 'VERTEX', ...}, ...}
+        .. code-block:: python
+
+            data = {
+                "data_A": {
+                    'values', [v_1, v_2, ...],
+                    'association': 'VERTEX'
+                    },
+                "data_B": {
+                    'values', [v_1, v_2, ...],
+                    'association': 'CELLS'
+                    },
+            }
 
         :return data_list: List of new Data objects
```

Comment:
wrong return format and missing period
```suggestion
        :return: List of new Data objects.
```